rsc/src/main/scala/rsc/outline/Synthesizer.scala

```diff
@@ -524,7 +502,10 @@ final class Synthesizer private (
         }
       }
       val ret = {
-        val params = tree.primaryCtor.get.paramss.headOption.getOrElse(Nil)
+        // we need to desugar because a single, implicit parameter list desugars
+        // into two parameters lists, the first one being empty:
+        // case class C(implicit val x: Int) desugars into case class C()(implicit val x: Int)
```

Comment:
I appreciate this explanation. Would it make sense to move this comment up (or write something complementary) since there are other cases above where you need to desugar?

Also, are there any other cases of desugaring that don't do what we want? (my guess is no, since desugaring is literally transforming source code into a form that more directly corresponds to compiler output, but just checking)

Reply:
I typically comment on cases where it is a bit surprising that a desugaring is happening (this was definitely surprising since we added a non-implicit parameter list). I think that if we continue to run into desugaring issues, we might want to have a higher-level comment at that point.